### PR TITLE
(GH-2358) Deprecate 'plugin_hooks' in favor of 'plugin-hooks'

### PR DESCRIPTION
**Pull request by beechtom**
This updates Bolt to issue a deprecation warning when a config file
includes the `plugin_hooks` option, which has been renamed to
`plugin-hooks`. When config is set for both of these options, Bolt
will issue another warning that `plugin_hooks` will be ignored.

!deprecation

* **Deprecate `plugin_hooks` in favor of `plugin-hooks`**
  ([#2358](#2358))

  The `plugin_hooks` configuration option has been deprecated in favor
  of `plugin-hooks`.

### CODE REVIEW

File: lib/bolt/project_manager/config_migrator.rb [hunk 1]
```diff
@@ -63,6 +64,34 @@ def migrate(config_file, project_file, inventory_file, backup_dir)
...
       end
+
+      private def update_options(project_file)
+        return true unless File.exist?(project_file)
+
+        @outputter.print_message("Updating project configuration options\n\n")
+        data     = Bolt::Util.read_yaml_hash(project_file, 'config')
+        modified = false
+
+        if data.key?('plugin_hooks') && !data.key?('plugin-hooks')
```
Comment (lucywyman) [inline lib/bolt/project_manager/config_migrator.rb:75]: This seems like it belongs in a different PR?

**Comment by lucywyman** [comment]
Should be able to remove the #2363 commit no...not sure why it's still showing up?

**Comment by beechtom** [comment]
Weird... I definitely rebased off main. 🤔

File: lib/bolt/config.rb [hunk 2]
```diff
@@ -567,7 +568,18 @@ def plugins
     end
 
     def plugin_hooks
-      @data['plugin_hooks']
+      if @data['plugin-hooks'].any?
```
Comment (lucywyman) [inline lib/bolt/config.rb:571]: I think this might be easier to follow as a single-level if statement:
```
if @data['plugin-hooks'].any? && @data['plugin_hooks'].any?
  issue warning and return plugin-hooks
elsif @data['plugin-hooks'].any?
  @data['plugin-hooks']
else
  @data['plugin_hooks']
end
```
```

**Comment by lucywyman** [comment]
Not sure you care, but there's a small typo in the release note:
```
The plugin_hooks configuration option has been deprecatedin favor
of plugin-hooks.
```

```
The plugin_hooks configuration option has been deprecated in favor
of plugin-hooks.
```

**Comment by lucywyman** [comment]
Oh no haha I think it got worse:
```
deprecatedi in
```
